pkg/util/status.go

```diff
@@ -62,6 +62,25 @@ func UpdateScaledObjectStatus(ctx context.Context, client runtimeclient.StatusCl
 	return TransformObject(ctx, client, logger, scaledObject, status, transform)
 }
 
+// UpdateTriggerAuthenticationStatus patches the given TriggerAuthentication/ClusterTriggerAuthentication with the updated status passed to it or returns an error.
+func UpdateTriggerAuthenticationStatus(ctx context.Context, client runtimeclient.StatusClient, logger logr.Logger, object runtimeclient.Object, status *kedav1alpha1.TriggerAuthenticationStatus) error {
```

Comment:
This is called only once from a single place in the code base. Think if we need to have it this way and having this as a public method.